test/apps/iodemo/io_demo.cc

```diff
@@ -1335,6 +1338,9 @@ class DemoClient : public P2pDemoCommon {
     }
 
     long get_num_uncompleted(const server_info_t& server_info) const {
+        assert(server_info.num_sent >= (server_info.num_completed[IO_READ] +
+                                        server_info.num_completed[IO_WRITE]));
+
```

Comment:
maybe calculate the result (as long) and assert it's >= 0 ?

Reply:
`long` can't protect from negative overflow, it's always >= 0
I implemented the same a bit different way in other PR https://github.com/openucx/ucx/pull/6338/files#diff-a12f7a78e9bc511657410e040a48b180f47969fbfcf2738f76136249fb13244fR1351

Reply:
isn't `long` signed?

Reply:
updated assert

Reply:
> isn't `long` signed?

you are right